src/main/java/seedu/tasklist/logic/parser/ListCommandParser.java

```diff
@@ -16,12 +14,14 @@ public ListCommand parse(String args) throws ParseException {
         boolean hasArgumentIndex = args.matches("(.+|[$&+,:;=?@#|'<>.^*()%!-])$"); //Has argument
         if (!hasArgumentIndex) {
             return new ListCommand(Index.fromZeroBased(0));
-        }
-        try {
-            Index index = ParserUtil.parseListIndex(args);
-            return new ListCommand(index);
-        } catch (ParseException pe) {
-            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, ListCommand.MESSAGE_USAGE), pe);
+        } else {
+            boolean argumentRange = args.trim().matches("^([1-9]|[1-4][0-9]|50)$");
+            if (argumentRange) {
+                Index index = ParserUtil.parseListIndex(args);
+                return new ListCommand(index);
+            } else {
+                throw new ParseException(ListCommand.MESSAGE_INDEX_NOT_IN_RANGE);
+            }
```

Comment:
Noicee. You may want to consider adding comments for the boolean `argumentRange` to tell the reader what the valid range is :)